### DISCUSSION TITLE
What does the `NonEmpty String` list contain in `namesFromLabelsWith`?

### DISCUSSION THREAD

**Discussion by sir4ur0n** [circuithub/rel8#279] [Q&A] [answered]
Function [namesFromLabelsWith](https://hackage.haskell.org/package/rel8-1.4.1.0/docs/Rel8.html#v:namesFromLabelsWith) takes as argument a function that receives a `NonEmpty String`, but the documentation does not explain what this list contains.
The documented example just takes the last value of that list to obtain the Haskell type.

I have fiddled with this function in a REPL but I always get back a list of exactly 1 element.

Can you explain (and ideally, add to the function documentation) what this list may contain, besides the last element always being the Haskell field name?

**Comment by shane-circuithub** [comment, marked as answer]
If you have a `Rel8able` like:

```haskell

data Foo f = Foo
  { a :: Column f Bool
  , b :: Column f Char
  }
  deriving (Generic, Rel8able)

data Bar f = Bar
  { foo1 :: Foo f
  , foo2 :: Foo f
  }
  deriving (Generic, Rel8able)
```

Then the argument passed to `namesFromLabelsWith` will contain things like `["foo1", "a"]`, `["foo1", "b"]`, `["foo2", "a"]`, `["foo2", "b"]`.

**Reply by sir4ur0n** [reply]
Thank you for the answer :pray: 

Shouldn't this be explained in the Haddock documentation? I think it trips up many developers while getting acquainted with Rel8 :sweat_smile: (Everyone in my team of Haskell developers wondered the same)

**Reply by ocharles** [reply]
Yea, it definitely should be. @sir4ur0n do you think you'd be able to contribute a little sentence doc PR? If not, I'll try and get to it soon.

**Reply by sir4ur0n** [reply]
I'm usually ok with contributing documentation to libraries, but I don't feel comfortable enough to do this one, because I don't fully understand the example above given by @shane-circuithub (https://github.com/circuithub/rel8/discussions/279#discussioncomment-7209620) :sweat_smile: 

And to the best of my knowledge, the pattern of nesting `Column` fields in a sub-type of a `Rel8able` type is not showcased in [the documentation](https://rel8.readthedocs.io/en/latest/index.html), so I feel like this would **also** deserve documentation.

I'd be happy to review any such PR and give feedback/suggestions though

**Reply by ocharles** [reply]
Ok, that's fair!
